source/material_model/melt_simple.cc

```diff
@@ -301,6 +311,21 @@ namespace aspect
                                                * out.densities[i] / this->get_timestep();
                   else
                     out.reaction_terms[i][c] = 0.0;
+
+                  // fill reaction rate outputs if the model uses operator splitting
+                  if (this->get_parameters().use_operator_splitting)
+                    {
+                      if (reaction_rate_out != NULL)
```

Comment:
What do you mean by that exactly? 

Right now, everywhere in the code where we need these outputs, they are created. If you write new code where you would need the outputs, but you do not call `create_additional_named_outputs`, you get a segmentation fault. 

If someone does not implement the `create_additional_named_outputs` function in the material model, they get the error from `compute_reactions()` that the material model does not implement the ReactionRateOutputs. 

Otherwise, if we wanted to change that statement to an assert, we would have to add a call to `create_additional_named_outputs` in all places where we call evaluate, and I thought the point of having them as a separate outputs was that they don't have to be computed in all places. We do the same for the melt outputs.